src/cartesian/Brush.tsx

```diff
@@ -290,10 +291,21 @@ export class Brush extends PureComponent<Props, State> {
   }
 
   handleDragEnd = () => {
-    this.setState({
-      isTravellerMoving: false,
-      isSlideMoving: false,
-    });
+    this.setState(
+      {
+        isTravellerMoving: false,
+        isSlideMoving: false,
+      },
+      () => {
+        const { endIndex, onDragEnd, startIndex } = this.props;
+        if (onDragEnd) {
+          onDragEnd({
+            endIndex,
+            startIndex,
+          });
+        }
```

Comment:
I just like conditional invocation:
```suggestion
          onDragEnd?.({
            endIndex,
            startIndex,
          });
```

Reply:
But totally a nit.

Reply:
Yes, lookd good. Commited this suggestion.